x-pack/solutions/security/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities_finding_flyout/vulnerability_finding_flyout.tsx

```diff
@@ -160,11 +165,80 @@ export const VulnerabilityFindingFlyout = ({
     { defaultMessage: 'Loading' }
   );
 
-  const vulnerabilityReference = vulnerability?.reference;
-
   const createVulnerabilityRuleFn = async (http: HttpSetup) =>
     await createDetectionRuleFromVulnerabilityFinding(http, vulnerabilityRecord);
 
+  const renderCveBadge = () => {
+    if (Array.isArray(vulnerability?.id) && vulnerability?.id.length > 1) {
+      return (
+        <EuiFlexItem grow={false}>
+          <EuiBadge
+            data-test-subj={VULNERABILITY_HEADER_CVE_BADGE}
+            color="accent"
+            css={css`
+              border-radius: ${euiTheme.border.radius.small};
+            `}
+          >
+            <FormattedMessage
+              id="xpack.csp.vulnerabilities.vulnerabilityFindingFlyout.cveTooltip"
+              defaultMessage="{value} More"
+              values={{ value: vulnerability?.id.length - 1 }}
+            />
+          </EuiBadge>
+        </EuiFlexItem>
+      );
+    }
+
+    return <></>;
+  };
+
+  const renderCves = () => {
+    if (!vulnerability?.id || vulnerability?.id.length === 0) {
+      return <></>;
+    }
+
+    const renderVulnerabilityId = () => {
+      const id = Array.isArray(vulnerability.id) ? vulnerability.id[0] : vulnerability.id;
```

Comment:
just an observation, but as we always render the first CVE below the title, it might highlight other CVEs in the title section than the one that was grouped, I found it a little confusing as I was expecting them to match, but not sure if that would also confuse users.

https://github.com/user-attachments/assets/d9e215ff-1ae9-4cf1-9bf1-0e8de4c3071c